src/app/features/userSlice.ts

```diff
@@ -0,0 +1,30 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { axiosConfig } from "app/utils";
+import axios from "axios";
+
+export const getAllUsers:any = createAsyncThunk("users/all",async ()=>{
+	try {
+		const {data}:any = await axios.get("/users",axiosConfig);
+		return data.users;
+	}catch(err:any) {
+		console.error({err});
+		return Promise.reject(err.response.data.message);
+	}
+});
+
+
+
+export const userState = createSlice({
+	name: "user",
+	initialState: [],
+	reducers: {},
+	extraReducers: {
+		[getAllUsers.fulfilled]: (state: any, action: any) => {
+			console.log({action});
+			state = action.payload;
+			return action.payload;
+		}
+	},
+});
+
```

Comment:
You can try to add an interface for the state instead of using any

Reply:
Yes, will do one refactor for that after some time. Thanks.